src/controllers/controller.bar.js

```diff
@@ -125,6 +125,57 @@ function computeFlexCategoryTraits(index, ruler, options) {
 	};
 }
 
+function parseFloatBar(arr, item, vScale, i) {
+	var startValue = vScale._parse(arr[0], i);
+	var endValue = vScale._parse(arr[1], i);
+	var min = Math.min(startValue, endValue);
+	var max = Math.max(startValue, endValue);
+	var barStart = min;
+	var barEnd = max;
+
+	if (Math.abs(min) > Math.abs(max)) {
+		barStart = max;
+		barEnd = min;
+	}
+
+	// Store `barEnd` (furthest away from origin) as parsed value,
+	// to make stacking straight forward
+	item[vScale.id] = barEnd;
+
+	item._custom = {
+		barStart: barStart,
+		barEnd: barEnd,
+		start: startValue,
+		end: endValue,
+		min: min,
+		max: max
+	};
+}
+
+function parseArrayOrPrimitive(meta, data, start, count) {
+	var iScale = this._getIndexScale();
+	var vScale = this._getValueScale();
+	var labels = iScale._getLabels();
+	var singleScale = iScale === vScale;
+	var parsed = [];
+	var i, ilen, item, entry;
+
+	for (i = start, ilen = start + count; i < ilen; ++i) {
+		entry = data[i];
+		item = {_index: i};
```

Comment:
I don't think `_index` is needed anymore

Reply:
removed